src/context/fix.rs

```diff
@@ -0,0 +1,57 @@
+use std::ops;
+
+use squalid::{IsEmpty, OptionExt};
+use tree_sitter_grep::tree_sitter::Node;
+
+#[derive(Default)]
+pub struct Fixer {
+    pending_fixes: Option<Vec<PendingFix>>,
+}
+
+impl Fixer {
+    pub fn replace_text(&mut self, node: Node, replacement: impl Into<String>) {
+        self.pending_fixes
+            .get_or_insert_with(Default::default)
+            .push(PendingFix::new(node.byte_range(), replacement.into()));
+    }
+
+    pub(crate) fn into_pending_fixes(self) -> Option<Vec<PendingFix>> {
+        self.pending_fixes
+    }
+
+    pub(crate) fn is_empty(&self) -> bool {
+        self.pending_fixes
+            .as_ref()
+            .is_none_or_matches(|pending_fixes| pending_fixes.is_empty())
+    }
+
+    pub fn remove_range(&mut self, range: ops::Range<usize>) {
+        self.pending_fixes
+            .get_or_insert_with(Default::default)
+            .push(PendingFix::new(range, Default::default()));
+    }
+
+    pub fn replace_text_range(&mut self, range: ops::Range<usize>, replacement: impl Into<String>) {
+        self.pending_fixes
+            .get_or_insert_with(Default::default)
+            .push(PendingFix::new(range, replacement.into()));
+    }
+}
+
+impl IsEmpty for Fixer {
```

Comment:
I think that I made a blanket implementation of `squalid`'s `NonEmpty` trait (or whatever I called it there) for anything that implements its `IsEmpty` trait (and implemented that for common types there)

So by implementing `IsEmpty` here I'm "opting in" to being able to use eg `.non_empty()` on this type